### PR TITLE
Paul/cleaner settings

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205443364901991

### CODE REVIEW

File: src/actions/LocalSettingsActions.ts [hunk 1]
```diff
@@ -149,15 +149,15 @@ export const writeSpendingLimits = async (account: EdgeAccount, spendingLimits:
...
-    .then(JSON.parse)
-    .catch(async () => {
-      // If Settings.json doesn't exist yet, create it, and return it
-      const defaults = asLocalAccountSettings({})
-      return await writeLocalSettings(account, defaults).then(() => defaults)
-    })
-    .then(settings => asLocalAccountSettings(settings))
+  try {
+    const text = await account.localDisklet.getText(LOCAL_SETTINGS_FILENAME)
+    const json = await JSON.parse(text)
```
Comment (samholmes) [inline src/actions/LocalSettingsActions.ts:154]: `JSON.parse` isn't async, so no need to await. 

Just FYI, `asLocalAccountSettings` could be a codec cleaner so we could `asLocalAccountSettings(text)` here if it were to use `asJSON`. Then on the write back to disk `wasLocalAccountSettings(data)`. But that's optional.

File: src/actions/LocalSettingsActions.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export const readLocalSettings = async (account: EdgeAccount): Promise<LocalAcco
   }
 }
 
-export const writeLocalSettings = async (account: EdgeAccount, settings: object) => {
+export const writeLocalSettings = async (account: EdgeAccount, settings: LocalAccountSettings) => {
   const text = JSON.stringify(settings)
```
Comment (samholmes) [inline src/actions/LocalSettingsActions.ts:164]: This is were you'd do the uncleaner to get the text: `wasLocalAccountSettings(settings)`. Again, optional.

**Review by samholmes — APPROVED** [review]
Approved with one minor change needed (an unnecessary await). An optional change to leverage `asJSON` as an additional commit.

File: src/actions/SettingsActions.tsx [hunk 3]
```diff
@@ -296,7 +296,7 @@ export const asSyncedAccountSettings = asObject({
   autoLogoutTimeInSeconds: asOptional(asNumber, 3600),
   defaultFiat: asOptional(asString, 'USD'),
   defaultIsoFiat: asOptional(asString, 'iso:USD'),
-  preferredSwapPluginId: asOptional(asString, ''),
+  preferredSwapPluginId: asOptional(asString),
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/actions/SettingsActions.tsx]: The value could always be undefined by the time it is used. We were converting empty string to Undefined before assigning it to the Settings object

File: src/actions/LocalSettingsActions.ts (hunk 1 above)
Comment (paullinator) [reply src/actions/LocalSettingsActions.ts:154]: Good catch on the async. Will fix. Do we really need an uncleaner if the data is always type checked before write

Comment (samholmes) [reply src/actions/LocalSettingsActions.ts:154]: Uncleaner is optional. Not necessary. It just moves the JSON serialization into the cleaner.

File: src/actions/SettingsActions.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/actions/SettingsActions.tsx]: Alright

**Pull request merged**